### PR TITLE
fix incident field laser and periodic field boundaries

### LINKED ISSUE #4450: incident field laser and periodic field boundaries are broken
As mentioned in PR #4449 the incident field plane wave laser is not working correctly if periodic boundaries are used.

# bunch example (step 700)
`./bin/picongpu -d 1 1 1 -g 128 1024 128 -s 2000 --periodic 1 0 1 --e_png.period 100 --e_png.axis yx --e_png.slicePoint 0.5 --e_png.folder pngElectronsYX`

e_png_yx_0 5_000700

You can see that boundaries are not periodic, the field must be equal in the transversal direction.

I tested activating the laser on all transversal boundaries too but this is not solving the issue :-(

### PR DESCRIPTION
**Pull request by psychocoderHPC**
fix #4450

Add a trait `MakePeriodicTransversalHuygensSurfaceContiguous` to extend the Huygens surface in transversal direction if they are periodic. If the surface is not extended to the outer boundaries the field will not be contiguous on the periodic boundaries.

### CODE REVIEW

File: include/picongpu/fields/incidentField/Solver.hpp [hunk 1]
```diff
@@ -188,7 +188,10 @@ namespace picongpu
                     typename T_CurlIncidentField,
                     typename T_FunctorIncidentField,
                     uint32_t T_axis>
-                inline void updateField(Parameters<T_axis> const& parameters, float_X const curlCoefficient)
+                inline void updateField(
+                    Parameters<T_axis> const& parameters,
+                    float_X const curlCoefficient,
+                    bool transversalContiguousHuygensSurface)
```
Comment (psychocoderHPC) [inline include/picongpu/fields/incidentField/Solver.hpp:196]: doxygen string missing

Comment (PrometheusPi) [reply include/picongpu/fields/incidentField/Solver.hpp:196]: great that you review your own code @psychocoderHPC 😉 
Could you add the doxygen string?

File: include/picongpu/fields/incidentField/Solver.hpp [hunk 2]
```diff
@@ -213,6 +216,52 @@ namespace picongpu
...
+                    const DataSpace<DIM3> periodic
+                        = Environment<simDim>::get().EnvironmentController().getCommunicator().getPeriodic();
+
+                    for(uint32_t d = 0u; d < DIM3; ++d)
+                    {
+                        bool isLaserAxisDirection = d == T_axis;
+
+                        // constexpr bool extendTransversalHuygensSurface
+                        //     =
+                        //     profiles::MakePeriodicTransversalHuygensSurfaceContiguous<T_FunctorIncidentField>::value;
```
Comment (PrometheusPi) [inline include/picongpu/fields/incidentField/Solver.hpp]: What is up with this dead/commented code?

**Review by PrometheusPi — DISMISSED** [review]
looks okay to me - @psychocoderHPC could you please state whether you still want to add the doxygen string and remove the commented code?

File: include/picongpu/fields/incidentField/Solver.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/picongpu/fields/incidentField/Solver.hpp]: deleted it, missed to delete it, dead code is always bad and must be removed.

**Comment by psychocoderHPC** [comment]
@PrometheusPi ready for review

File: include/picongpu/fields/incidentField/Solver.hpp (hunk 1 above)
Comment (PrometheusPi) [reply include/picongpu/fields/incidentField/Solver.hpp:196]: I assume not needed

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**